examples/llama.py

```diff
@@ -16,102 +16,89 @@
 
 MAX_CONTEXT = getenv("MAX_CONTEXT", 4096)
 
-# **** files and arguments ****
+# calculating params:
+# traditionally, the MLP in the transformer architecture has hidden_dim = dim*4 [arxiv/1706.03762, 3.3]
+# however, Llama uses SwiGLU. in order to preserve param count to original transformer arch, hidden_dim must be = 2/3 * (dim*4) [arxiv/2002.05202]
+# for models using MQA (n_kv_heads != n_heads), preserving param count means hidden dim must be further multiplied by 1.3 [arxiv/2307.09288, A.2.1]
 MODEL_PARAMS = {
   "1": {
     "7B": {
-      "args": {"dim": 4096, "multiple_of": 256, "n_heads": 32, "n_layers": 32, "norm_eps": 1e-06, "vocab_size": 32000},
-      "files": 1,
```

Comment:
can you keep the trailing comma? it's helpful to keep the change minimal when adding/removing new entries